VideoLocker/src/main/java/org/edx/mobile/view/adapters/CourseOutlineAdapter.java

```diff
@@ -83,18 +84,27 @@ public View getRowView(int position, View convertView, ViewGroup parent) {
             convertView.setBackgroundResource(R.color.white);
         }
         final ViewHolder viewHolder = (ViewHolder)convertView.getTag();
-        if( row.component.isLastChild() ){
+
+        if ( nextRow == null ){
             viewHolder.halfSeparator.setVisibility(View.GONE);
+            viewHolder.wholeSeparator.setVisibility(View.VISIBLE);
         } else {
-            viewHolder.halfSeparator.setVisibility(View.VISIBLE);
+            viewHolder.wholeSeparator.setVisibility(View.GONE);
+            boolean isLastChildInBlock = !row.component.getParent().getId().equals( nextRow.component.getParent().getId());
+            if( isLastChildInBlock ){
+                viewHolder.halfSeparator.setVisibility(View.GONE);
+            } else {
+                viewHolder.halfSeparator.setVisibility(View.VISIBLE);
+            }
```

Comment:
Perhaps we should remove the separator entirely at the end of the last item? I think the real issue was that the `component.isLastChild()` was not returning true when there were items ahead of it that had been filtered out.